### PR TITLE
chore: Add ChainReplaceAttr

### LINKED ISSUE #15: Add `ChainReplaceAttr(...func([]string, slog.Attr) slog.Attr) func([]string, slog.Attr) slog.Attr`
Panics if called without args.

If one of chained functions returns zero `slog.Attr` or attr with `.Value.Kind() == slog.KindGroup` then it should skip next functions in a chain and return this value.

### PR DESCRIPTION
**Pull request by salamandra19**
Close #15

### CODE REVIEW

File: replace_attr.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package slogx
+
+import "log/slog"
+
+func ChainReplaceAttr(fn ...func([]string, slog.Attr) slog.Attr) func([]string, slog.Attr) slog.Attr {
```
Comment (powerman) [inline replace_attr.go:5]: `fns` or `fs`

File: replace_attr.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	if len(fn) == 0 {
+		panic("arguments required")
+	}
+
+	return func(g []string, a slog.Attr) slog.Attr {
+		attr := a
```
Comment (powerman) [inline replace_attr.go:11]: Remove.

**Review by powerman — APPROVED** [review]

File: replace_attr_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+			}
+			return a
+		}
+	)
+
+	t.Panic(func() { slogx.ChainReplaceAttr() })
+
+	fn := slogx.ChainReplaceAttr(modifyAttrValue, modifyAttrKey, returnZeroAttr, ignoreAfterZeroAttr)
+	t.DeepEqual(fn([]string{}, slog.Attr{Key: id, Value: slog.IntValue(325)}), slog.Attr{Key: userID, Value: slog.StringValue("REDACTED")})
+	t.DeepEqual(fn([]string{"g"}, slog.Attr{Key: id, Value: slog.IntValue(325)}), slog.Attr{Key: userID, Value: slog.StringValue("REDACTED")})
```
Comment (salamandra19) [inline replace_attr_test.go:50]: I added the call with not empty group. But I am not sure you mean exactly such test. Group gets lost. @powerman

Comment (powerman) [reply replace_attr_test.go:50]: Groups does not lost, they just didn't affect chained function's output. Test should check is chained functions will get same groups in their argument.

**Pull request merged**